### PR TITLE
CSD simulation experiments: parameter selection + domain addition

### PR DESCRIPTION
**Pull request by jjc2718**
In this PR, we wanted to try two experiments using simulated data to test the model implementation from #61:

* Fix the latent space dimension (we refer to as _k_) for the simulations, and see if a model using the "true"/"correct" latent space dimension outperforms models using other latent space dimensions
* Add domains one by one to the training dataset, and see if model performance on a held-out domain improves (we would expect this to be the case if we're correctly distinguishing "common" from "domain-specific" signal)

For the first experiment, we don't really see that the fixed latent space dimension of the model affects performance much at all:

image

And for the second experiment, we do see that sometimes adding domains has the expected effect (this is for a simulated dataset with 50 features and 6 domains):

n6_p50_corr1_scaleNone_points

But with only 25 features, the pattern is not so clear:

n6_p25_corr1_scaleNone_points

I'm not too concerned about the first result, but will do some follow-up on the second result in a future PR.

### CODE REVIEW

**Review by arielah — APPROVED** [review]
Looks good! Sorry for the delay in reviewing.

File: 09_simulations/fit_models.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    train_ridge,
+    train_rf,
+    train_mlp,
+    train_linear_csd
+)
+
+def get_prob_metrics(y_train, y_test, y_pred_train, y_pred_test):
+    """Get metrics for predicted probabilities.
+
+    y_pred_train and y_pred_test should be continuous; true values are binary.
```
[the remark this answers is not in the document]
Comment (jjc2718) [reply 09_simulations/fit_models.py:17]: I'm trying to feed it the output of logistic regression (`y_pred_train` and `y_pred_test` are the predicted probabilities of having a 1 label for the train and test sets respectively), and I use that to calculate performance metrics like AUROC/AUPR. I'll elaborate a bit in the documentation to make this clearer.

File: 09_simulations/models.py [hunk 2]
```diff
@@ -280,142 +273,3 @@ def train_linear_csd(X_train,
     return cv_pipeline
 
 
-def get_prob_metrics(y_train, y_test, y_pred_train, y_pred_test):
```
[the remark this answers is not in the document]
Comment (jjc2718) [reply 09_simulations/models.py:283]: I just wanted to keep this file as small as it made sense to. In my mind `models.py` just has the code for fitting the models and making predictions given some train/test data, and `fit_models.py` has the code for doing everything "around" the model fitting without actually fitting them: e.g. preprocess/split the data, and post-process the predictions by calculating performance metrics using the true labels.

I'll add a bit more detail to the file docstrings about this (and maybe change the file names since I now realize `fit_models.py` doesn't actually fit the models).

File: 09_simulations/scripts/simulate_k_params.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Script to run simulate_k.ipynb for several different sets of simulation
+# parameters using papermill, writing the resulting plots to the
+# simulation_plots directory.
+
+# Note that this must be run from the parent directory
```
[the remark this answers is not in the document]
Comment (jjc2718) [reply 09_simulations/scripts/simulate_k_params.sh:7]: Good idea, I'll do that!
